.github/workflows/deployment.yml

```diff
@@ -0,0 +1,126 @@
+    name: Modify version and build package for TestPyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: Set environment variable for version suffix
+        run: echo "MRPROVERSIONSUFFIX=.pre$(date +%s)" >> $GITHUB_ENV
+
+      - name: Build package with version suffix
+        run: |
+          python -m pip install --upgrade build
+          python -m build
+
+      - name: Store TestPyPi distribution
+        uses: actions/upload-artifact@v4
+        with:
+          name: testpypi-package-distribution
+          path: dist/
+
+  testpypi-deployment:
+    name: Deploy to TestPyPI
+    needs:
+      - build-testpypi-package
+    runs-on: ubuntu-latest
+
+    environment:
+      name: testpypi
+      url: https://test.pypi.org/p/mrpro
+
+    permissions:
+      id-token: write
+
+    steps:
+      - name: Download TestPyPi distribution
+        uses: actions/download-artifact@v4
+        with:
+          name: testpypi-package-distribution
+          path: dist/
+
+      - name: Publish to TestPyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
+          verbose: true
+
+  test-install-from-testpypi:
+    name: Test installation from TestPyPI
+    needs:
+      - testpypi-deployment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: Install MRpro from TestPyPI
+        run: |
+          python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ mrpro
+
+  build-pypi-package:
+    name: Build package for PyPI
+    runs-on: ubuntu-latest
+    needs:
+      - test-install-from-testpypi
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: Build package
+        run: |
+          python -m pip install --upgrade build
+          python -m build
+
+      - name: Store PyPi distribution
+        uses: actions/upload-artifact@v4
+        with:
+          name: pypi-package-distribution
+          path: dist/
+
+  pypi-deployment:
+    name: Deploy to PyPI
+    needs:
+      - test-install-from-testpypi
+    runs-on: ubuntu-latest
+
+    environment:
+      name: pypi
+      url: https://pypi.org/project/mrpro
+
+    permissions:
+      id-token: write
+
+    steps:
+      - name: Download PyPi distribution
+        uses: actions/download-artifact@v4
+        with:
+          name: pypi-package-distribution
+          path: dist/
+
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        if: github.ref == 'refs/heads/main'
```

Comment:
can we move this up such that the whole job only runs if the condition is true?

Reply:
maybe this would also resolve the wrong deployed to pypi message..

Reply:
Yeah, this is what I suggested above already:

> EDIT: maybe we want to move the if: github.ref == 'refs/heads/main' condition from step to job level? Else GitHub Actions will always report that we successfully deployed to PyPi although we didn't 😁

Reply:
I did put it at the step level during development to ensure all other steps work

Reply:
I did not see the edit :D